rest/src/main/java/fi/otavanopisto/pyramus/rest/controller/StudentController.java

```diff
@@ -537,6 +550,44 @@ public void deleteLodgingPeriod(StudentLodgingPeriod period) {
 
     return result;
   }
-
+  
+  /* Checks if logged user is guidance counselor of specific student */
+  public boolean amIGuidanceCounselor(Long studentId, StaffMember staffMember) {
+    
+    if (staffMember == null) {
+      return false;
+    }
+    
+    Student student = findStudentById(studentId);
+    // List of student's user groups
+    List<StudentGroup> studentGroups = studentGroupDAO.listByStudentAndUserAndGuidanceGroupAndArchived(student, staffMember, true, false);
+    
+    return !studentGroups.isEmpty();
+  }
+  
+  public ContactLogAccess resolveContactLogAccess(Student student) {
+    User loggedUser = sessionController.getUser();
+    
+    if (loggedUser.getRole().equals(Role.ADMINISTRATOR)) {
+      return ContactLogAccess.ALL;
+    } 
+    StaffMember staffMember = userController.findStaffMemberById(loggedUser.getId());
+    
+    if (staffMember == null) {
+      return ContactLogAccess.NONE;
+    } 
+    
+    Boolean amIGuidanceCounselor = amIGuidanceCounselor(student.getId(), staffMember);
+    
+    if (amIGuidanceCounselor) {
+      return ContactLogAccess.ALL;
+    } else {
+      if (!staffMember.getRole().equals(Role.CLOSED)) {
```

Comment:
`CLOSED` role check could be moved to the `staffMember == null` check; no need to even figure out whether we are a guidance councelor if our role is `CLOSED` anyway.